test.js

```diff
@@ -69,10 +95,106 @@ test('it supports multiple arguments', t => {
   source.emit(1, 'qu');
 
   t.deepEqual(
-    sink.getReceivedData(), 
+    sink.getReceivedData(),
     ['foo','bar','baz','qu'],
     'sink gets seed and subsequent data'
   );
 
   t.end();
 });
+
+test('it queues sync completion', t => {
+  const seededSrc = startWith('a', 'b', 'c')(of('d'));
```

Comment:
Not 100% sold on using `of`. Isn't that just abstracting away doing...

```javascript
seededSrc.emit(1, 'd');
seededSrc.emit(2);
```

...which would make the test more readable?

Reply:
On the other hand, if we keep it, we should refactor the later test that uses `fromIter` to use `of` instead, and get rid of `forIter`.

That might be worth doing anyway, but for this test I like the readability that my verbose code ☝️ would provide.

Reply:
`of` and manual mock emits are not the same, difference is that `of` emits its values **within** greet, manual mocks would emit them after both sides (source, sink) are fully "set up"

also `of` and `fromIter` are not interchangeable, former is listenable, latter is pullable, so I've used both in different tests to test different semantics

Reply:
also not sure how valuable this test will be when I refactor this operator to `concat` semantics (not yet using actual `concat` operator) 🤔

Reply:
> difference is that `of` emits its values **within** greet

Just so I understand; you mean after handshake but in the same tick?

Reply:
Not exactly, but similar. Both flows would happen completely within a single tick, but with manual mock emits we would achieve such flow:
```js
1. source gets sink
2. source gives a talkback to sink
3. greet ends (imagine using pipe - we are now after pipe is finished)
4. we send data to sink through mock
5. we send completion to sink through mock
```

with `of ` its slightly different
```js
1. source gets sink
2. source gives a talkback to sink
3. we send data to sink through of
4. we send completion to sink through of
5. greet ends (we are now after pipe is finished)
```

With somehow wrongly implemented operator we could screw either of those scenarios (although more likely the `of` scenario can be wrongly handled as its more "special").

Reply:
Doesn't the `of` scenario violate the Callbag spec?

> * A callbag MUST NOT be delivered data before it has been greeted

Reply:
I don't think so (please discuss further if you think otherwise, I might be wrong about this), `of` [sends `0` first](https://github.com/Andarist/callbag-of/blob/b273e229582b1166a35f91b071b292f28bfc5a54/src/index.js#L7) & [data (`1`) afterward](https://github.com/Andarist/callbag-of/blob/b273e229582b1166a35f91b071b292f28bfc5a54/src/index.js#L13-L15), so I believe callbag (sink) is already greeted when the data is delivered to it.

Weirdness of this is that this all happens before `pipe` has a chance to return.

Reply:
No, wait, I was just confused by the "greet ends" wording. The handshake is of course completed as we send 0 and the talkback to the sink. Sorry, again, vacation brain. :P